demo/alice-blueprint-commit.hs

```diff
@@ -0,0 +1,107 @@
+{-# LANGUAGE ExtendedDefaultRules #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+import Control.Lens (asIndex, to, (&), (^.), (^?))
+import Data.Aeson
+import qualified Data.Aeson.Key as Key
+import Data.Aeson.Lens
+import Data.ByteString.Lazy (ByteString)
+import qualified Data.ByteString.Lazy as LBS
+import Data.Maybe (fromJust)
+import Data.String
+import Data.Text (Text)
+import qualified Data.Text.Lazy as Text
+import Data.Text.Lazy.Encoding (decodeLatin1)
+import Shh
+
+getAddress :: IO ByteString
+getAddress = do
+  exe
+    "cardano-cli"
+    "address"
+    "build"
+    "--payment-verification-key-file"
+    "devnet/credentials/alice-funds.vk"
+    "--testnet-magic"
+    "42"
+    |> capture
+
+getUTxO :: ByteString -> IO (Either String Value)
+getUTxO address = do
+  k <-
+    exe
+      "cardano-cli"
+      "query"
+      "utxo"
+      "--socket-path"
+      "devnet/node.socket"
+      "--address"
+      address
+      "--testnet-magic"
+      "42"
+      "--output-json"
+      |> capture
+  pure $ eitherDecode k
+
+buildTransaction :: ByteString -> ByteString -> IO ()
+buildTransaction i o =
+  exe
+    "cardano-cli"
+    "conway"
+    "transaction"
+    "build-raw"
+    "--tx-in"
+    i
+    "--tx-out"
+    (o <> "+500000")
+    "--fee"
+    "0"
+    "--out-file"
+    "tx.json"
+
+blueprintTx :: Text -> Key -> Text -> Value
+blueprintTx address utxo cborHex =
+  object
+    [ "blueprintTx"
+        .= object
+          [ "cborHex" .= String cborHex
+          , "description" .= ""
+          , "type" .= "Tx ConwayEra"
+          ]
+    , "utxo"
+        .= object
+          [ utxo
+              .= object
+                [ "address" .= String address
+                , "datum" .= Null
+                , "datumhash" .= Null
+                , "inlinedatum" .= Null
+                , "referenceScript" .= Null
+                , "value" .= object [
+                    "lovelace" .= Number 800000
+                  ]
+                ]
+          ]
+    ]
+
+decodeTxFile :: FilePath -> IO (Either String Value)
+decodeTxFile fp = do
+  x <- LBS.readFile "tx.json"
+  pure $ eitherDecode x
+
+main = do
+  address <- getAddress
+  utxo <- either error id <$> getUTxO address
+  print utxo
+  let utxoKey = utxo ^. members . asIndex
+  buildTransaction (Data.String.fromString . Key.toString $ utxoKey) address
+  tx <- either error id <$> decodeTxFile "tx.json"
+  print tx
+  let cborHex = tx ^? key "cborHex"
+  let (String cborHexBS) = fromJust cborHex
```

Comment:
No need to write to a file, you should be able to use the output of `transaction build-raw` directly as the `blueprintTx` later (you can trick the cardano-cli to use `--out-file /dev/stdout`).